packages/core/event.ts

```diff
@@ -57,11 +57,25 @@ export function getEventName(eventNameObj: EventNameObjectType, forAttendeeView
     .replaceAll("{Event type title}", eventNameObj.eventType)
     .replaceAll("{Scheduler}", attendeeName)
     .replaceAll("{Organiser}", eventNameObj.host)
+    .replaceAll("{Organiser first name}", eventNameObj.host.split(" ")[0])
     .replaceAll("{USER}", attendeeName)
     .replaceAll("{ATTENDEE}", attendeeName)
     .replaceAll("{HOST}", eventNameObj.host)
     .replaceAll("{HOST/ATTENDEE}", forAttendeeView ? eventNameObj.host : attendeeName);
 
+  if (
+    eventNameObj.bookingFields &&
+    eventNameObj.bookingFields.name &&
+    typeof eventNameObj.bookingFields.name === "object" &&
+    !Array.isArray(eventNameObj.bookingFields.name) &&
+    eventNameObj.bookingFields.name.firstName
+  ) {
+    dynamicEventName = dynamicEventName.replaceAll(
+      "{Scheduler first name}",
+      eventNameObj.bookingFields.name.firstName.toString()
+    );
+  }
+
```

Comment:
NIT
```suggestion
  if (
    eventNameObj.bookingFields?.name &&
    typeof eventNameObj.bookingFields.name === "object" &&
    !Array.isArray(eventNameObj.bookingFields.name) &&
    eventNameObj.bookingFields.name.firstName
  ) {
    dynamicEventName = dynamicEventName.replaceAll(
      "{Scheduler first name}",
      eventNameObj.bookingFields.name.firstName.toString()
    );
  }

```

and creating a func isObj to check typeof eventNameObj.bookingFields.name === "object" &&
    !Array.isArray(eventNameObj.bookingFields.name) &&

Reply:
![Screenshot 2024-05-13 at 2 19 10 PM](https://github.com/calcom/cal.com/assets/74371312/f02213da-d570-4298-a80f-5184da18d5bd)

Cannot shake this error